src/middleware/middleware.js

```diff
@@ -58,30 +58,36 @@ export default class Middleware {
    * Runs a middleware chain for a specific action.
    * @param {string} action - The action to run.
    * @param {Function} callback - The callback function.
+   * @param {Array<any>} params - The action params.
    * @public
    * @returns {void}
    */
-  run(action: string, callback: Function): void {
+  run(action: string, callback: Function, ...params: Array<any>): void {
     this._logger.debug('Start middleware chain for action ' + action);
     let middlewares = this._middlewares.get(action);
-    this._executeMiddleware(middlewares, () => {
-      this._logger.debug('Finish middleware chain for action ' + action);
-      callback();
-    });
+    this._executeMiddleware(
+      middlewares,
+      () => {
+        this._logger.debug('Finish middleware chain for action ' + action);
+        callback();
+      },
+      params
+    );
   }
 
   /**
    * Executes all the middlewares one by one.
    * @param {Array<Function>} middlewares - The middlewares for a specific action.
    * @param {Function} callback - The callback function.
+   * @param {Array<any>} params - The action params.
    * @private
    * @returns {void}
    */
-  _executeMiddleware(middlewares: Array<Function>, callback: Function): void {
+  _executeMiddleware(middlewares: Array<Function>, callback: Function, params: Array<any>): void {
     const composition = middlewares.reduceRight(
       // eslint-disable-next-line no-unused-vars
       (next, fn) => v => {
-        fn(next);
+        params?.length > 0 ? fn(...params, next) : fn(next);
```

Comment:
```suggestion
        params?.length ? fn(...params, next) : fn(next);
```

Reply:
what is next in this case ? how does the middleware distinguish between next and params if it comes after params ?

Reply:
next have to be the last param